lib/puppet/provider/cisco_interface/cisco.rb

```diff
@@ -201,6 +216,74 @@ def ipv4_addr_mask_set
     end
   end
 
+  def stp_mst_cost
+    return @property_hash[:stp_mst_cost] if @resource[:stp_mst_cost].nil?
+    if @resource[:stp_mst_cost][0] == :default &&
+       @property_hash[:stp_mst_cost] == @interface.default_stp_mst_cost
+      return [:default]
+    else
+      @property_hash[:stp_mst_cost]
+    end
+  end
+
+  def stp_mst_cost=(should_list)
+    should_list = @interface.default_stp_mst_cost if should_list[0] == :default
+    # check for overlapping arrays in should_list
+    PuppetX::Cisco::Utils.fail_array_overlap(should_list)
```

Comment:
Did you consider calling this method in the type instead?  Also, what does the error message look like when you run `puppet agent -t` for this?